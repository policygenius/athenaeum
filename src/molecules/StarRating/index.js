import React from 'react'; import PropTypes from 'prop-types';
import classnames from 'classnames';
import times from 'lodash/times';
import Icon from 'atoms/Icon';
import styles from './star_rating.module.scss';

function StarRating(props) {
  const {
    className,
    size,
    rating,
  } = props;

  const classes = classnames(
    styles['star-rating'],
    styles.stars,
    styles[`stars-${rating}`],
    className,
  );

  return (
    <span className={classes}>
      {
        times(5, i =>
          <Icon
            key={i}
            className={classnames(
              styles.star,
              {
                [styles['filled-in']]: i < rating
              }
            )}
            icon={i < rating ? 'star2' : 'starLight'}
            width={size}
            height={size}
            renderSVGDOM
          />
        )
      }
    </span>
  );
}


StarRating.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * Add custom inline style
   */
  style: PropTypes.object,

  /**
   * Pixel amount for height and width of stars.
   */
  size: PropTypes.string,

  /**
   * Number of stars that are filled in.
   */
  rating: PropTypes.number.isRequired,
};

StarRating.defaultProps = {
  height: '12px',
  // Place any default props here.
};

export default StarRating;
