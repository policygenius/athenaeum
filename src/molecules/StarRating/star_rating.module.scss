.star-rating {
  // Place your component specific scss here.
  line-height: 0;
}

.star {
  margin-left: rem-calc(6.4px);

  &:first-of-type {
    margin-left: 0;
  }
}

span[class*='stars'] {
  color: color('primary-3');
}
// When the stars switch to SVG icons,
// will want to change this to svg fill.

.stars-1 {
  span:nth-of-type( 1n + 2 ) svg path {
    fill: color('neutral-4');
  }
}

.stars-2 {
  span:nth-of-type( 1n + 3 ) svg path {
    fill: color('neutral-4');
  }
}

.stars-3 {
  span:nth-of-type( 1n + 4 ) svg path {
    fill: color('neutral-4');
  }
}

.stars-4 {
  span:nth-of-type( 1n + 5 ) svg path {
    fill: color('neutral-4');
  }
}
// hack to get the `*=stars` class to apply
// to insurance companies with 5 stars
.stars-5 {
  color: color('primary-3');
}

