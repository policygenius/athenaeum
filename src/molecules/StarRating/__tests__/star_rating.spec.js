import React from 'react';
//import sinon from 'sinon';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import StarRating from 'molecules/StarRating';

describe('<StarRating />', () => {
  // let defaultProps;
  //
  // beforeEach(() => {
  //   defaultProps = {
  //     // Add default props here
  //   };
  // });

  it('renders correctly', () => {
    const wrapper = mount(<StarRating />);

    expect(wrapper.type()).toEqual(StarRating);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<StarRating />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
