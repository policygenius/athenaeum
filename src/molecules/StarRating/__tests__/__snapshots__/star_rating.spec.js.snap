// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating /> renders correctly 1`] = `
<span
  className="star-rating stars stars-4"
>
  <span
    className="icon-wrapper star filled-in"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="star2"
      className="img"
      src="https://static.policygenius.com/svgs/star2.svgz"
      title="star2"
    />
  </span>
  <span
    className="icon-wrapper star filled-in"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="star2"
      className="img"
      src="https://static.policygenius.com/svgs/star2.svgz"
      title="star2"
    />
  </span>
  <span
    className="icon-wrapper star filled-in"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="star2"
      className="img"
      src="https://static.policygenius.com/svgs/star2.svgz"
      title="star2"
    />
  </span>
  <span
    className="icon-wrapper star filled-in"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="star2"
      className="img"
      src="https://static.policygenius.com/svgs/star2.svgz"
      title="star2"
    />
  </span>
  <span
    className="icon-wrapper star"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="starLight"
      className="img"
      src="https://static.policygenius.com/svgs/starLight.svgz"
      title="starLight"
    />
  </span>
</span>
`;
